jenkins-slave

```diff
@@ -45,5 +45,10 @@ else
 		URL="-url $JENKINS_URL"
 	fi
 
-	exec java $JAVA_OPTS -cp /usr/share/jenkins/slave.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL "$@"
+	if [ ! -z "$JNLP_PROTOCOL_OPTS" ]; then
```

Comment:
isn't this the opposite of what you want to do?  if JNLP_PROTOCOL_OPTS is not set then set it

Another option is to just add to the Dockerfile

    ENV JNLP_PROTOCOL_OPTS="-Dorg.jenkinsci.remoting.engine.JnlpProtocol3.disabled=true"